packages/langium/test/grammar/grammar-util.test.ts

```diff
@@ -86,12 +87,47 @@ describe('Find name assignment in parser rules', () => {
         expect(nameAssignment).toBe(undefined);
     });
 
+    test('Should be able to infer data type rule for cross ref', () => {
```

Comment:
I'd propose to name this test "Should be able to find a named assignment by Inferred Type".